internal/app/users/service.go

```diff
@@ -303,11 +303,24 @@ func (us *service) ListUsers(ctx context.Context, reqData dto.ListUsersReq) (res
 		return
 	}
 
+	userId := ctx.Value(constants.UserId)
+
 	var users []dto.UserDetails
 
 	for _, dbUser := range dbResp {
-		user := mapDbUserToUserListResp(dbUser)
-		users = append(users, user)
+		if reqData.Self {
+			if dbUser.Id != userId {
+				user := mapDbUserToUserListResp(dbUser)
+				users = append(users, user)
+			}
+		} else {
+			user := mapDbUserToUserListResp(dbUser)
+			users = append(users, user)
+		}
+	}
```

Comment:
if reqData.Self && dbUser.Id == userID {
    continue
}

user := mapDbUserToUserListResp(dbUser)
users = append(users, user)

This should work :)